sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/KeyVaultSecretIdentifier.java

```diff
@@ -16,11 +16,47 @@
 public final class KeyVaultSecretIdentifier {
     private final String secretId, vaultUrl, name, version;
 
-    private KeyVaultSecretIdentifier(String secretId, String vaultUrl, String name, String version) {
-        this.secretId = secretId;
-        this.vaultUrl = vaultUrl;
-        this.name = name;
-        this.version = version;
+    /**
+     * Create a new {@link KeyVaultSecretIdentifier} from a given secret identifier.
+     *
+     * <p>Valid examples are:
+     *
+     * <ul>
+     *     <li>https://{key-vault-name}.vault.azure.net/secrets/{secret-name}</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/secrets/{secret-name}/pending</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/secrets/{secret-name}/{unique-version-id}</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/deletedsecrets/{deleted-secret-name}</li>
+     * </ul>
+     *
+     * @param secretId The secret identifier to extract information from.
+     *
+     * @throws IllegalArgumentException if the given identifier is {@code null} or an invalid Key Vault Secret
+     * identifier.
+     */
+    public KeyVaultSecretIdentifier(String secretId) {
+        if (secretId == null) {
+            throw new IllegalArgumentException("secretId cannot be null");
+        }
+
+        try {
+            final URL url = new URL(secretId);
+            // We expect an identifier with either 2 or 3 path segments: collection + name [+ version]
+            final String[] pathSegments = url.getPath().split("/");
+
+            if ((pathSegments.length != 3 && pathSegments.length != 4) // More or less segments in the URI than expected.
+                || !"https".equals(url.getProtocol()) // Invalid protocol.
+                || (!"secrets".equals(pathSegments[1]) && !"deletedsecrets".equals(pathSegments[1])) // Invalid collection.
+                || ("deletedsecrets".equals(pathSegments[1]) && pathSegments.length == 4)) { // Deleted items do not include a version.
```

Comment:
Ditto

Reply:
Left [a comment](https://github.com/Azure/azure-sdk-for-java/pull/20357#discussion_r607427937) about this above.